src/state/internal/createQueryStore.ts

```diff
@@ -851,7 +851,18 @@ export function getQueryKey<TParams extends Record<string, unknown>>(params: TPa
   return JSON.stringify(
     Object.keys(params)
       .sort()
-      .map(key => params[key])
+      .map(key => {
+        const value = params[key];
+        // Convert Sets to sorted arrays for consistent serialization
+        if (value instanceof Set) {
+          return Array.from(value).sort();
```

Comment:
i merged a getConsistentArray helper recently, maybe we could abstract a getConsistentMap to match